src/vtp/script_libs/cast_ballot_lib.py

```diff
@@ -0,0 +1,334 @@
+        if tally == "plurality":
+            print(f"- This is a {tally} tally")
+            print(
+                f"- The voting is for {max_votes} open position(s)/choice(s) - "
+                "only that number of selections can be choosen."
+            )
+        else:
+            print(
+                f"- This is a {tally} tally with {max_votes} open position(s)/choice(s).  "
+                f"Regardless up to {len(choices)} selections can be rank choosen."
+            )
+
+        # Need to print the choices first up front
+        count = 0
+        for choice in choices:
+            print(f"  [{count}] {choice}")
+            count += 1
+
+        def validate_multichoice(text):
+            """Will validate the space separated user input choice string"""
+            selections = text.split()
+            choice_max = len(choices)
+            errors = []
+            validated_selections = []
+            for num in selections:
+                if not num.isnumeric():
+                    errors.append(f"The supplied choice ({num}) is not a number")
+                    continue
+                num = int(num)
+                if num < 0 or num > choice_max:
+                    errors.append(
+                        f"The supplied choice ({num}) is not a valid selection "
+                        f"(must be between 0 and {choice_max - 1} inclusive)"
+                    )
+                    continue
+                if tally == "plurality" and len(selections) > choice_max:
+                    errors.append(
+                        f"This contest is limited to at most {max_votes} selection(s): "
+                        f"you supplied {len(selections)}"
+                    )
+                    continue
+                #            import pdb; pdb.set_trace()
+                if num in validated_selections:
+                    errors.append(
+                        f"The selection {num} was supplied more than once.  "
+                        "Each selection can only be supplied once."
+                    )
+                    continue
+                validated_selections.append(num)
+            if errors:
+                err_string = "\n".join(errors)
+                raise ValueError(
+                    "Warning - you selections have the following errors\n"
+                    f"{err_string}"
+                )
+            # if still here, set the selection
+            try:
+                # Since it is possible to self adjudicate a contest, always
+                # explicitly clear the selection before adding
+                the_ballot.clear_selection(the_contest)
+                for sel in validated_selections:
+                    the_ballot.add_selection(the_contest, sel)
+            # pylint: disable=broad-except
+            except Exception:
+                # blow out of the internal pyinputplus try/catch
+                traceback.print_exc()
+                sys.exit(1)
+
```

Comment:
This won't work with a REST API, and it can't be tested. For now, leave it as is, but at some point it will need to be refactored so it can work outside of a CLI.

Reply:
Yes!  FWIIW, for the spring demo by design this part will not be called/executed because the JS will be handling this (as discussed last week).  Actually, when the support for that endpoint is added, this may naturally go away - or at least I will try to make it go away then.